moveit_planners/chomp/chomp_motion_planner/src/chomp_planner.cpp

```diff
@@ -52,16 +52,38 @@ bool ChompPlanner::solve(const planning_scene::PlanningSceneConstPtr& planning_s
                          moveit_msgs::MotionPlanDetailedResponse& res) const
 {
   if (!planning_scene)
+  {
     ROS_ERROR_STREAM("No planning scene initialized.");
-  else
+    res.error_code.val = moveit_msgs::MoveItErrorCodes::FAILURE;
+    return false;
+  }
+
+  if (req.start_state.joint_state.position.empty())
+  {
+    ROS_ERROR_STREAM("Starting state is empty");
+    res.error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_ROBOT_STATE;
+    return false;
+  }
+
+  if (not planning_scene->getRobotModel()->satisfiesPositionBounds(req.start_state.joint_state.position.data()))
```

Comment:
this is actually older than c++11 but microsoft decided not to support it because it would break compatibility with C code were it could have been defined using macros. Besides who wants to follow standards anyway?

Theres also 'and', 'or', ... turning it into a slightly less ugly language. See http://en.cppreference.com/w/cpp/language/operator_alternative

Reply:
I find them cute in their own weird way :)